common/djangoapps/student/helpers.py

```diff
@@ -814,3 +819,120 @@ def user_has_passing_grade_in_course(enrollment):
     except AttributeError:
         pass
     return False
+
+
+def _prepare_instructor_object(staff, marketing_root_url):
+    """
+    Prepare staff object according to data required in email template
+    """
+    instructor = {
+        'name': f"{staff['given_name']} {staff['family_name']}",
+        'profile_image_url': staff['profile_image_url'],
+        'organization_name': staff['position']['organization_name'] if staff['position'] else '',
+        'bio_url': f"{marketing_root_url}/bio/{staff['slug']}"
+    }
+    return instructor
+
+
+def get_instructors(course_run, marketing_root_url):
+    """
+    Get course instructors in such a way if instructors count is odd then first object from list will set in
+    first_instructor and remaining will append in instructors list.
+    If instructor count is even then first object will be null and all objects will append in instructors list.
+    All the above mentioned logic is required for Braze course enrollment email design template.
+    """
+    first_instructor = {}
+    instructors = []
+    if 'staff' in course_run:
+        if len(course_run['staff']) % 2 != 0:
+            staff = course_run['staff'][0]
+            first_instructor.update(_prepare_instructor_object(staff, marketing_root_url))
+
+        even_instructors = course_run['staff'] if len(course_run['staff']) % 2 == 0 else course_run['staff'][1:]
+        for instructor in even_instructors:
+            obj = _prepare_instructor_object(instructor, marketing_root_url)
+            instructors.append(obj)
+
+    return first_instructor, instructors
+
+
+def _prepare_date_block(block, block_date, user_timezone):
+    """
+    Prepare date block which include assignment related data fot this date
+    """
+    timezone_offset = get_time_zone_offset(user_timezone, block_date)
+    block = {
+        'title': block['title'] or '',
+        'assignment_type': block['assignment_type'] or '',
+        'assignment_count': 0,
+        'link': block['link'] or '',
+        'date': block_date,
+        'due_date': block_date.strftime("%a, %b %d, %Y"),
+        'due_time': (f'{block_date.strftime("%H:%M %p")} GMT{timezone_offset}' if block['assignment_type'] else '')
+    }
+    return block
+
+
+def get_course_dates_for_email(user, course_id, request):
+    """
+    Getting nearest dates from today one would be before today and one
+    would be after today.
+    """
+    user_timezone_locale = user_timezone_locale_prefs(request)
+    user_timezone = timezone(user_timezone_locale['user_timezone'] or str(UTC))
+
+    course = get_course_with_access(user, 'load', course_id, check_if_enrolled=False)
+    date_blocks = get_course_date_blocks(course, user, request, include_access=True, include_past_dates=True)
+    date_blocks = [block for block in date_blocks if not isinstance(block, TodaysDate)]
```

Comment:
Can we remove this check `if not isinstance(block, TodaysDate)`? It will include today's date automatically including all the due assignments on that day and you won't need to calculate it later on at L926-930 explicitly.

Reply:
And by doing this we might be able to improve the dates extraction logic to sort the date blocks on the date field and then pick the today, one previous, and one after item of it instead of looping on it.